SonarQube.Bootstrapper/BuildAgentUpdater.cs

```diff
@@ -99,6 +117,81 @@ public bool CheckBootstrapperApiVersion(string versionFilePath, Version bootstra
             return false;
         }
 
+
+        public void InjectLoaderTargets(ILogger logger)
+        {
+            WarnOnGlobalTargetsFile(logger);
+            InternalCopyTargetsFile(logger);
+        }
+
+        private static void InternalCopyTargetsFile(ILogger logger)
+        {
+            string sourceTargetsPath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), LoaderTargetsName);
+            Debug.Assert(File.Exists(sourceTargetsPath), String.Format("Could not find the {0} file in the current director", LoaderTargetsName));
+
+            CopyIfDifferent(sourceTargetsPath, DestinationDirs, logger);
+        }
+
+        private static void CopyIfDifferent(string sourcePath, IEnumerable<string> destinationDirs, ILogger logger)
+        {
+            string sourceContent = GetReadOnlyFileContent(sourcePath);
+            string fileName = Path.GetFileName(sourcePath);
+
+            foreach (string destinationDir in destinationDirs)
+            {
+                string destinationPath = Path.Combine(destinationDir, fileName);
+
+                if (!File.Exists(destinationPath))
+                {
+                    Directory.CreateDirectory(destinationDir); // creates all the directories in the path if needed
+                    File.Copy(sourcePath, destinationPath, overwrite: false);
+                    logger.LogMessage(Resources.INFO_CopiedTargets, fileName, destinationDir);
+                }
+                else
+                {
+                    string destinationContent = GetReadOnlyFileContent(destinationPath);
+
+                    if (!String.Equals(sourceContent, destinationContent, StringComparison.Ordinal))
+                    {
+                        File.Copy(sourcePath, destinationPath, overwrite: true);
+                        logger.LogMessage(Resources.INFO_CopiedTargets, fileName, destinationDir);
+                    }
+                }
+            }
+        }
+
+        private static string GetReadOnlyFileContent(string path)
```

Comment:
The important thing here is that opening the file is done with the flag FileShare.Read. According to MSDN it is possible for multiple processes to read from the same file if this flag is set.